src/tree/VirtualMachineTreeItem.ts

```diff
@@ -97,6 +97,7 @@ export class VirtualMachineTreeItem extends AzureTreeItem {
             if (multiDelete) { ext.outputChannel.appendLog(deleting); }
 
             const failedResources: string[] = await deleteAllResources(this.root, this.resourceGroup, resourcesToDelete);
+            await ext.tree.refresh(context, this.parent);
```

Comment:
You shouldn't need to use refresh at all. This is supposed to be handled by `deleteTreeItem` which does it in a much more efficient way because it doesn't need to clear the whole cache

https://github.com/microsoft/vscode-azuretools/blob/main/ui/src/tree/AzExtTreeItem.ts#L111

Can you find a way to work with that?

Reply:
In that case, I think the best alternative is to use a warning message rather than an error and just track failures in telemetry.